### PR TITLE
Make editor attributes panel dynamic

### PR DESCRIPTION
**Pull request by burieberry**
- Editor panel display changes depending on what the user clicks on the rendered template
- User can return to main panel ("global attributes") by clicking elsewhere on page
- Stop trying to show overlay for hidden fields
- Adjust colors and z-indices on tools

### CODE REVIEW

File: packages/tools/addon/styles/cs-field-overlays.css [hunk 1]
```diff
@@ -37,3 +42,15 @@
 .cs-field-overlay.passthrough > * {
   pointer-events: auto;
 }
+
+/* Overriding style attribute that sets `visibility` to `hidden`.
+   This is so that when user can exit focus on field by
+   clicking elsewhere on the page */
+.cs-field-overlay.focused > .liquid-child {
+  visibility: visible !important;
```
Comment (ef4) [inline packages/tools/addon/styles/cs-field-overlays.css]: This is likely to cause animations to flicker. The liquid-child is only supposed to be hidden temporarily, during the start of an animation. If it's staying hidden, that's a bug, and we should fix that instead.

File: packages/tools/addon/templates/components/cs-active-composition-panel.hbs [hunk 2]
```diff
@@ -9,62 +9,25 @@
...
-              </div>
-            {{/let}}
-          {{/each}}
-        </div>
-
-        {{#each renderedFields key="id" as |fieldMark|}}
-          {{! template-lint-disable attribute-indentation }}
+  {{#if openedFieldId}}
+    <div class="cs-active-composition-panel--parameters">
+      {{#each renderedFields key="id" as |fieldMark|}}
```
Comment (ef4) [inline packages/tools/addon/templates/components/cs-active-composition-panel.hbs]: This template file was already at or beyond the maximum acceptable complexity. The copy-paste duplication introduced by this PR pushes it way beyond understandable. So we need to factor the shared pieces out into one or more separate components.

File: packages/tools/addon/styles/cs-field-overlays.css (hunk 1 above)
Comment (burieberry) [reply packages/tools/addon/styles/cs-field-overlays.css]: Yes, something is off with the scrim behavior, I'll investigate further.

File: packages/tools/addon/templates/components/cs-field-overlays.hbs [hunk 3]
```diff
@@ -6,20 +6,25 @@
       mark.model.content
       mark.model.caption
       (eq mark.id tools.openedFieldId)
+      (eq mark.id tools.detailedViewFieldId)
       (cs-field-editor-for mark.model.content mark.model.name variant="inline")
-    as |fieldName content caption isOpenedField editorName|}}
+    as |fieldName content caption isOpenedField isDetailedViewField editorName|}}
       {{#let (is-component editorName) as |hasInlineEditor|}}
+        <div hidden=true onclick={{action "openField" mark}} data-test-overlay-button={{dasherize fieldName}}>
```
Comment (burieberry) [inline packages/tools/addon/templates/components/cs-field-overlays.hbs]: What's a better way to do this? Trying to click on the field overlay below doesn't work during testing so I had to add a hidden `div` with the same click action on it.

File: packages/tools/addon/templates/components/cs-active-composition-panel.hbs (hunk 2 above)
Comment (burieberry) [reply packages/tools/addon/templates/components/cs-active-composition-panel.hbs]: Any better now?

File: packages/tools/addon/templates/components/cs-field-overlays.hbs (hunk 3 above)
Comment (jenweber) [reply packages/tools/addon/templates/components/cs-field-overlays.hbs]: Do you have any ideas why the click doesn't work in testing?

I would add an Esc key action, for both keyboard nav and testing. In testing you'd have to still set the focus on the overlay though. See "closing the dialog" on https://www.nomensa.com/blog/2014/how-improve-accessibility-overlay-windows-part-1

File: packages/tools/addon/templates/components/cs-active-composition-panel.hbs (hunk 2 above)
Comment (jenweber) [reply packages/tools/addon/templates/components/cs-active-composition-panel.hbs]: @ef4 can you take another look at this?

**Comment by burieberry** [comment]
I'm closing this since the changes in tools package are currently irrelevant. We can reopen in the future if necessary.

**Pull request closed without merging**